src/lib/libnodefs.js

```diff
@@ -112,65 +112,84 @@ addToLibrary({
       }
       return newFlags;
     },
-
+    getattr(func, node) {
+      var stat = NODEFS.tryFSOperation(func);
+      if (NODEFS.isWindows) {
+        // node.js v0.10.20 doesn't report blksize and blocks on Windows. Fake
+        // them with default blksize of 4096.
+        // See http://support.microsoft.com/kb/140365
+        if (!stat.blksize) {
+          stat.blksize = 4096;
+        }
+        if (!stat.blocks) {
+          stat.blocks = (stat.size+stat.blksize-1)/stat.blksize|0;
+        }
+        // Windows does not report the 'x' permission bit, so propagate read
+        // bits to execute bits.
+        stat.mode |= (stat.mode & {{{ cDefs.S_IRUGO }}}) >> 2;
+      }
+      return {
+        dev: stat.dev,
+        ino: node.id,
+        mode: stat.mode,
+        nlink: stat.nlink,
+        uid: stat.uid,
+        gid: stat.gid,
+        rdev: stat.rdev,
+        size: stat.size,
+        atime: stat.atime,
+        mtime: stat.mtime,
+        ctime: stat.ctime,
+        blksize: stat.blksize,
+        blocks: stat.blocks
+      };
+    },
+    // Common code for both node and stream setattr
+    // For node getatrr:
+    //  - arg is a native path
+    //  - chmod, utimes, truncate are fs.chmodSync,  fs.utimesSync,  fs.truncateSync
+    // For stream getatrr:
+    //  - arg is a native file descriptor
+    //  - chmod, utimes, truncate are fs.fchmodSync, fs.futimesSync, fs.ftruncateSync
+    setattr(arg, node, attr, chmod, utimes, truncate, stat) {
```

Comment:
I can't say I love this method of adding the opaque `arg` along with all the opts for it.. but I can't think of a nicer way, and I think its better than duplicating the logic.